Mathlib/Data/Bitvec/Core.lean

```diff
@@ -310,48 +311,77 @@ attribute [local simp] Nat.zero_add Nat.add_zero Nat.one_mul Nat.mul_one Nat.zer
 
 -- mul_left_comm
 theorem toNat_append {m : ℕ} (xs : Bitvec m) (b : Bool) :
-    Bitvec.toNat (xs++ₜb ::ᵥ nil) = Bitvec.toNat xs * 2 + Bitvec.toNat (b ::ᵥ nil) :=
-  by
+    Bitvec.toNat (xs++ₜb ::ᵥ nil) = Bitvec.toNat xs * 2 + Bitvec.toNat (b ::ᵥ nil) := by
   cases' xs with xs P
-  simp [bits_to_nat_to_list]; clear P
-  unfold bits_to_nat List.foldl
+  simp [bitsToNat_toList]; clear P
+  unfold bitsToNat List.foldl
   -- generalize the accumulator of foldl
   generalize h : 0 = x;
-  conv in add_lsb x b => rw [← h]; clear h
+  conv in addLsb x b =>
+    rw [← h];
   simp
+  clear h
   induction' xs with x xs generalizing x
   · simp
-    unfold List.foldl add_lsb
+    unfold addLsb
+    simp
     simp [Nat.mul_succ]
```

Comment:
Are you sure you need this new intermediate `simp`?

Reply:
If memory serves me right, it is necessary to work. I can check this again later (3 hours or so)

Reply:
If you need it, please replace it with the output of `simp?`.

Reply:
Okay. I'll do that (tomorrow)

Reply:
I got a suggestion from simp? and inserted this suggested `simp only ....`. The proof works.

Reply:
I deleted the new `simp`, it wasn't needed.